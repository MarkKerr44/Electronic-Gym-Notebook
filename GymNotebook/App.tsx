import React from 'react';
import 'react-native-reanimated';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import OpeningScreen from './OpeningScreen';

const Stack = createNativeStackNavigator();

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Opening" screenOptions={{ headerShown: false }}>
        <Stack.Screen name="Opening" component={OpeningScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
