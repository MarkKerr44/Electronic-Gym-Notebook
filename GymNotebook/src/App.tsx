import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import OpeningScreen from './OpeningScreen';
import SignUpScreen from './SignUpScreen';
import SignInScreen from './SignInScreen';
import { DashboardScreen } from './DashboardScreen';
import WorkoutSelectionScreen from './WorkoutSelectionScreen';
import PoseEstimatorScreen from './PoseEstimatorScreen';
import ExerciseLibraryScreen from './ExerciseLibraryScreen';
import StartWorkoutScreen from './StartWorkout';
import { RoutineSetupScreen } from './RoutineSetupScreen';
import WorkoutHistoryScreen from './WorkoutHistoryScreen';
import WeightTrackerScreen from './WeightTrackerScreen';
import SettingsScreen from './SettingsScreen';
import MyWorkoutsScreen from './MyWorkoutsScreen';
import UserProfileScreen from './UserProfileScreen';
import ExerciseDetailScreen from './exercise/[exerciseId]';
import CreateWorkoutScreen from './createWorkout';
import { WorkoutDetailsScreen } from './WorkoutDetails';

const Stack = createNativeStackNavigator();

function App() {
  return (
    <GestureHandlerRootView style={{ flex: 1 }} testID="app-root">
      <NavigationContainer>
        <Stack.Navigator initialRouteName="OpeningScreen">
          <Stack.Screen name="OpeningScreen" component={OpeningScreen} options={{ headerShown: false }} />
          <Stack.Screen name="SignUpScreen" component={SignUpScreen} options={{ title: 'Sign Up' }} />
          <Stack.Screen name="SignInScreen" component={SignInScreen} options={{ title: 'Sign In' }} />
          <Stack.Screen name="DashboardScreen" component={DashboardScreen} options={{ title: 'Dashboard' }} />
          <Stack.Screen name="WorkoutSelectionScreen" component={WorkoutSelectionScreen} options={{ title: 'Select Workout' }} />
          <Stack.Screen name="PoseEstimatorScreen" component={PoseEstimatorScreen} options={{ title: 'Pose Estimator' }} />
          <Stack.Screen name="ExerciseLibraryScreen" component={ExerciseLibraryScreen} options={{ title: 'Exercise Library' }} />
          <Stack.Screen name="StartWorkout" component={StartWorkoutScreen} options={{ title: 'Start Workout' }} />
          <Stack.Screen name="RoutineSetupScreen" component={RoutineSetupScreen} options={{ title: 'Routine Setup' }} />
          <Stack.Screen name="WorkoutHistoryScreen" component={WorkoutHistoryScreen} options={{ title: 'Workout History' }} />
          <Stack.Screen name="WeightTrackerScreen" component={WeightTrackerScreen} options={{ title: 'Weight Tracker' }} />
          <Stack.Screen name="SettingsScreen" component={SettingsScreen} options={{ title: 'Settings' }} />
          <Stack.Screen name="MyWorkoutsScreen" component={MyWorkoutsScreen} options={{ title: 'My Workouts' }} />
          <Stack.Screen name="UserProfileScreen" component={UserProfileScreen} options={{ title: 'User Profile' }} />
          <Stack.Screen name="ExerciseDetails" component={ExerciseDetailScreen} options={{ title: 'Exercise Detail' }} />
          <Stack.Screen name="createWorkout" component={CreateWorkoutScreen} options={{ title: 'Create Workout' }} />
          <Stack.Screen name="WorkoutDetails" component={WorkoutDetailsScreen} options={{ title: 'Workout Details' }} />
        </Stack.Navigator>
      </NavigationContainer>
    </GestureHandlerRootView>
  );
}

export default App;
