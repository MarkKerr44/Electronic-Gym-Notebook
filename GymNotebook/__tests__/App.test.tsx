import React from 'react';
import { render, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import renderer from 'react-test-renderer';
import App from '../src/App';
import { beforeEach, describe, it, expect, jest } from '@jest/globals';

const TestWrapper = ({ children }: { children: React.ReactNode }) => (
  <NavigationContainer>
    {children}
  </NavigationContainer>
);

describe('App Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders without crashing', async () => {
    await waitFor(() => {
      const { getByTestId } = render(
        <TestWrapper>
          <App />
        </TestWrapper>
      );
      expect(getByTestId('app-root')).toBeTruthy();
    });
  });

  it('contains NavigationContainer', async () => {
    await waitFor(() => {
      const component = renderer.create(
        <TestWrapper>
          <App />
        </TestWrapper>
      );
      expect(component.root.findByType(NavigationContainer)).toBeTruthy();
    });
  });

  it('matches the snapshot', async () => {
    await waitFor(() => {
      const tree = renderer
        .create(
          <TestWrapper>
            <App />
          </TestWrapper>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('renders navigation structure correctly', async () => {
    const { getByTestId } = render(
      <TestWrapper>
        <App />
      </TestWrapper>
    );

    await waitFor(() => {
      expect(getByTestId('app-root')).toBeTruthy();
    });
  });

  it('handles authentication state', async () => {
    const { getByTestId } = render(
      <TestWrapper>
        <App />
      </TestWrapper>
    );

    await waitFor(() => {
      expect(getByTestId('app-root')).toBeTruthy();
    });
  });
});