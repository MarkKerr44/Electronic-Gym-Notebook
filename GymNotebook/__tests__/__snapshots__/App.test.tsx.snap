// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
  testID="app-root"
/>
`;
